Warning about “writing to an object of type 'struct NoiseParams' with no trivial copy-assignment” in file src/noise.cpp when compiling

##### Issue type
<!-- Pick one below and delete others -->
 - Build issue
 
##### Minetest version
<!--
Paste Minetest version between quotes below
If you are on a devel version, please add git commit hash
You can use `minetest --version` to find it.
-->
```
$ git rev-parse HEAD
111f1dc9c5f336f5b4f12c15940dc4cc2cbe7f82
```

##### OS / Hardware
<!-- General information about your hardware and operating system -->
Operating system: Linux 4.20.6-arch1-1-ARCH x86_64
CPU: Intel(R) Core(TM) i7-4790K CPU @ 4.00GHz

GNU Make 4.2.1
cmake version 3.13.4

##### Summary
When compiling Minetest as described in the readme at one point the following warning appears.

```
/home/dirk/test/minetest/src/noise.cpp: In constructor 'Noise::Noise(NoiseParams*, irr::s32, irr::u32, irr::u32, irr::u32)':
/home/dirk/test/minetest/src/noise.cpp:432:29: warning: 'void* memcpy(void*, const void*, size_t)' writing to an object of type 'struct NoiseParams' with no trivial copy-assignment; use copy-assignment or copy-initialization instead [-Wclass-memaccess]
  memcpy(&np, np_, sizeof(np));
                             ^
In file included from /home/dirk/test/minetest/src/noise.cpp:27:
/home/dirk/test/minetest/src/noise.h:103:8: note: 'struct NoiseParams' declared here
 struct NoiseParams {
        ^~~~~~~~~~~
```

##### Steps to reproduce
1. Compile the latest available version (when writing this it’s 111f1dc9c5f336f5b4f12c15940dc4cc2cbe7f82)
2. Watch the build output

--- a/src/client/mapblock_mesh.cpp
+++ b/src/client/mapblock_mesh.cpp
@@ -419,7 +419,14 @@ static void getNodeVertexDirs(const v3s16 &dir, v3s16 *vertex_dirs)
 	u8 idx = (dir.X + 2 * dir.Y + 3 * dir.Z) & 7;
 	idx = (idx - 1) * 4;
 
+#if defined(__GNUC__) && !defined(__clang__)
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wclass-memaccess"
+#endif
 	memcpy(vertex_dirs, &vertex_dirs_table[idx], 4 * sizeof(v3s16));
+#if defined(__GNUC__) && !defined(__clang__)
+#pragma GCC diagnostic pop
+#endif
 }
 
 static void getNodeTextureCoords(v3f base, const v3f &scale, const v3s16 &dir, float *u, float *v)

--- a/src/noise.cpp
+++ b/src/noise.cpp
@@ -424,7 +424,7 @@ float NoisePerlin3D(NoiseParams *np, float x, float y, float z, s32 seed)
 
 Noise::Noise(NoiseParams *np_, s32 seed, u32 sx, u32 sy, u32 sz)
 {
-	memcpy(&np, np_, sizeof(np));
+	np = *np_;
 	this->seed = seed;
 	this->sx   = sx;
 	this->sy   = sy;

